pkg/cstorvolume/v1alpha1/build.go

```diff
@@ -0,0 +1,281 @@
+	if len(namespace) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing namespace"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Namespace = namespace
+	return b
+}
+
+// WithAnnotations merges existing annotations if any
+// with the ones that are provided here
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing annotations"),
+		)
+		return b
+	}
+
+	if b.cstorvolume.object.Annotations == nil {
+		return b.WithAnnotationsNew(annotations)
+	}
+
+	for key, value := range annotations {
+		b.cstorvolume.object.Annotations[key] = value
+	}
+	return b
+}
+
+// WithAnnotationsNew resets existing annotations if any with
+// ones that are provided here
+func (b *Builder) WithAnnotationsNew(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: no new annotations"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newannotations := map[string]string{}
+	for key, value := range annotations {
+		newannotations[key] = value
+	}
+
+	// override
+	b.cstorvolume.object.Annotations = newannotations
+	return b
+}
+
+// WithLabels merges existing labels if any
+// with the ones that are provided here
+func (b *Builder) WithLabels(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing labels"),
+		)
+		return b
+	}
+
+	if b.cstorvolume.object.Labels == nil {
+		return b.WithLabelsNew(labels)
+	}
+
+	for key, value := range labels {
+		b.cstorvolume.object.Labels[key] = value
+	}
+	return b
+}
+
+// WithLabelsNew resets existing labels if any with
+// ones that are provided here
+func (b *Builder) WithLabelsNew(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: no new labels"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newlbls := map[string]string{}
+	for key, value := range labels {
+		newlbls[key] = value
+	}
+
+	// override
+	b.cstorvolume.object.Labels = newlbls
+	return b
+}
+
+// WithTargetIP sets the target IP address field of
+// CStorVolume with provided arguments
+func (b *Builder) WithTargetIP(targetip string) *Builder {
+	if len(targetip) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing targetip"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Spec.TargetIP = targetip
+	return b
+}
+
+// WithCapacity sets the Capacity field of CStorVolume with provided arguments
+func (b *Builder) WithCapacity(capacity string) *Builder {
+	if len(capacity) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolume object: missing capacity"),
+		)
+		return b
+	}
+	b.cstorvolume.object.Spec.Capacity = capacity
```

Comment:
instead of string, can capacity used as k8s Quantity object? cc: @prateekpandey14 @AmitKumarDas 
(or) that becomes a schema change and needs upgrade for existing CVs?